Fix build against pango >= 1.57.2, which now defines the PangoRenderer
autoptr cleanup itself (pango-renderer.h), colliding with the local
definition and failing with "redefinition of glib_autoptr_clear_
PangoRenderer". Backport of the guard from mutter main.

--- a/clutter/clutter/pango/clutter-pango-private.h	2026-07-03 13:44:46.066663751 +0200
+++ b/clutter/clutter/pango/clutter-pango-private.h	2026-07-03 13:44:46.067663751 +0200
@@ -79,7 +79,9 @@
                           ClutterColorState *color_state,
                           ClutterColorState *target_color_state);
 
+#if !PANGO_VERSION_CHECK(1, 57, 2)
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (PangoRenderer, g_object_unref)
+#endif
 
 static inline PangoDirection
 clutter_text_direction_to_pango_direction (ClutterTextDirection dir)
